File: src/main/java/com/medicalsystem/repository/PatientRepository.java
package com.medicalsystem.repository;

import com.medicalsystem.model.Form;
import com.medicalsystem.model.Patient;
import org.springframework.data.jpa.repository.JpaRepository;

import java.util.List;

public interface PatientRepository extends JpaRepository<Patient, Long> {
    List<Patient> findAllByForm(Form form);
    boolean existsByPatientId(String patientId);
}

File: src/main/kotlin/com/medicalsystem/repository/PatientRepository.kt
package com.medicalsystem.repository

import com.medicalsystem.domain.Patient
import com.medicalsystem.domain.template.Form
import org.springframework.data.jpa.repository.JpaRepository

interface PatientRepository : JpaRepository<Patient, String> {

    fun findAllByForm(form: Form): List<Patient>
}